### DISCUSSION TITLE
difference inference result detect.py and other method

### DISCUSSION THREAD

**Discussion by pieter-willems** [ultralytics/yolov5#8889] [Q&A] [answered]
hi everyone! I have been trying to create a counter using yolov5. the model that i have created has a high map and should be adequate for this. When I use the detect.py script that is provided with yolov5 I get an almost perfect inference result. However since I need access to every frame of the video and build some code around it, I have been using the code that I have included. The inference result that I get from this code when compared to detect.py is really bad though and not good enough to make a counter with. I was wondering how I could get a better inference result using this method or maybe how I could integrate detect.py in my code to make inferences that way?

import torch
import cv2
import numpy as np

def main():
	model = torch.hub.load('ultralytics/yolov5', 'custom',path='best.pt', force_reload=True)

         cap= cv2.VideoCapture('example.mp4')
         #loop for reading frames from video
         while cap.isOpened():
		 ret, frame = cap.read()
		 results=model(frame)
                 cv2.imshow('video',np.squeeze(results.render()))
		
		#exit using q key
                if cv2.waitKey(10) & 0xff == ord('q'):
                break
	  cap.release()
          cv2.destroyAllWindows()

if __name__ == '__main__':
    main()

**Comment by glenn-jocher** [comment, marked as answer]
@pieter-willems user error. cv2 loads in BGR. Follow the PyTorch Hub Tutorial cv2 Usage examples.

## YOLOv5 Tutorials

* [Train Custom Data](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://github.com/ultralytics/yolov5/wiki/Tips-for-Best-Training-Results)&nbsp; ☘️  RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Roboflow for Datasets, Labeling, and Active Learning](https://github.com/ultralytics/yolov5/issues/4975)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://github.com/ultralytics/yolov5/issues/475)
* [PyTorch Hub](https://github.com/ultralytics/yolov5/issues/36)&nbsp; ⭐ NEW
* [TFLite, ONNX, CoreML, TensorRT Export](https://github.com/ultralytics/yolov5/issues/251) 🚀
* [Test-Time Augmentation (TTA)](https://github.com/ultralytics/yolov5/issues/303)
* [Model Ensembling](https://github.com/ultralytics/yolov5/issues/318)
* [Model Pruning/Sparsity](https://github.com/ultralytics/yolov5/issues/304)
* [Hyperparameter Evolution](https://github.com/ultralytics/yolov5/issues/607)
* [Transfer Learning with Frozen Layers](https://github.com/ultralytics/yolov5/issues/1314)&nbsp; ⭐ NEW
* [Architecture Summary](https://github.com/ultralytics/yolov5/issues/6998)&nbsp; ⭐ NEW

Good luck 🍀 and let us know if you have any other questions!

**Reply by pieter-willems** [reply]
This solved my issue. Thank you for helping!
